internal/errors/is.go

```diff
@@ -6,6 +6,16 @@ import (
 	"github.com/lib/pq"
 )
 
+// IsDomainError returns a boolean indicating whether the error is
+// a boundary domain error
+func IsDomainError(err error) bool {
```

Comment:
Helper function to avoid needing to do the four lines each time, open to alternatives

Reply:
If you look at Convert inside error.go it will return either the domain error with the proper type or it will return nil so instead of using errors.As you could use

if dErr := errors.Convert(err); err != nil { ... }

Maybe we should update the IsX() methods below to use Convert.

Reply:
Hum... I may change my mind about using errors.Convert in this use-case.  Convert will only work if the type == errors.Err, and errors.As will work for this use-case and for wrapped domain errors.  I guess, it depends on how you want this to work

Reply:
Isn't a wrapped errors.Err still a errors.Err?  Maybe Convert() should handle unwrapping it for you.

Reply:
Convert() could easily unwrap it and that's probably the right solution.  Right now, it's just attempting a cast to *Err